### PR TITLE
Exp2 op using lookup tables

### LINKED ISSUE #28: feat: Supporting `Exp2` Operator
This issue aims to add support for the `Exp2` operator in LuminAIR. 

# Implementation Requirements

## 1. `PrimitiveCompiler` Support

- Add `Exp2` operator handling in the [PrimitiveCompiler](https://github.com/gizatechxyz/LuminAIR/blob/70ef5eb1c49c3c126cd3be07942ac17ade6e026b/crates/graph/src/op/prim.rs#L121)
- Follow the pattern established for other operators like `Add` and `Mul`
- Ensure proper connection with the AIR component

## 2. AIR Component Implementation

- Implement the `Exp2` AIR component in [air/src/components/mod.rs](https://github.com/gizatechxyz/LuminAIR/blob/70ef5eb1c49c3c126cd3be07942ac17ade6e026b/crates/air/src/components/mod.rs#L171)
- Define appropriate trace generation and constraints for the `Exp2` operation

## 3. Value-Level Implementation
Implement fixed-point `Exp2` operation in the [NumerAIR](https://github.com/gizatechxyz/NumerAir) repository for:
- `FixedBaseField` backend
- `FixedPackedBaseField` backend

## 4. Testing

- Add unit tests verifying the correctness of the `Exp2` operation in NumerAIR repository
- Add integration tests demonstrating the full flow (build graph → compile → prove → verify) in [graph/src/tests/ops.rs](https://github.com/gizatechxyz/LuminAIR/blob/70ef5eb1c49c3c126cd3be07942ac17ade6e026b/crates/graph/src/tests/ops.rs)

## 5. Documentation

- Update the operators documentation in [docs/concepts/operators.mdx](https://github.com/gizatechxyz/LuminAIR/blob/master/docs/concepts/operators.mdx) to change the status of `Exp2` from ⏳ to ✅
- Add any specific notes about the implementation if necessary

## References

Similar operator implementations like [Add](https://github.com/gizatechxyz/LuminAIR/blob/70ef5eb1c49c3c126cd3be07942ac17ade6e026b/crates/air/src/components/add.rs) and [Mul](https://github.com/gizatechxyz/LuminAIR/blob/70ef5eb1c49c3c126cd3be07942ac17ade6e026b/crates/air/src/components/mul.rs) can be used as references

## Acceptance Criteria

- The `Exp2` operator works correctly for all valid inputs
- A proof can be generated and verified for computational graphs using the `Exp2` operator
- All tests pass, including edge cases
- Documentation is updated to reflect the new implementation status
- The implementation follows the project's code style and architectural patterns

### PR DESCRIPTION
**Pull request by bruhhgnik**
fixes #28 
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: crates/graph/src/tests/expansions.rs [hunk 1]
```diff
@@ -265,22 +265,6 @@ fn test_large_expansion_factors() -> Result<(), Box<dyn std::error::Error>> {
...
-
-        // Apply unary operation, then expand
-        let processed = base.sin();
-        let expanded = processed.expand(2, 3);
-
-        // Use expanded result in binary operation
-        let other = cx.tensor((2, 2, 3)).set(random_vec_rng(12, &mut rng, false));
-        expanded * other
-    })
-}
```
[the remark this answers is not in the document]
Comment (bruhhgnik) [reply crates/graph/src/tests/expansions.rs:283]: ah yes must be a mistake

**Comment by raphaelDkhn** [comment]
#74 is based on this PR. We can close this. Thank you @bruhhgnik !

**Comment by raphaelDkhn** [comment]
@all-contributors add @bruhhgnik for code
